### PR TITLE
feat: Add global admin role (no-changelog)

### PR DESCRIPTION
**Pull request by valya**
Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

File: packages/cli/src/databases/migrations/common/1700571993961-AddGlobalAdminRole.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+			name: 'admin',
+			scope: 'global',
+		});
+	}
+
+	async down({ escape, runQuery }: MigrationContext) {
+		const roleTableName = escape.tableName('role');
+		const userTableName = escape.tableName('user');
+
+		const adminRoleIdResult: Array<{ id: number }> = await runQuery(
```
Comment (ivov) [inline packages/cli/src/databases/migrations/common/1700571993961-AddGlobalAdminRole.ts:17]: Looks like we can pass the result type into the generic. Same below.

File: packages/cli/src/databases/migrations/common/1700571993961-AddGlobalAdminRole.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		const adminRoleId = adminRoleIdResult[0]?.id;
+		if (!adminRoleId) {
+			// Couldn't find admin role. It's a bit odd but it means we don't
+			// have anything to do.
+			return;
+		}
+
+		const memberRoleId = memberRoleIdResult[0]?.id;
+		if (!memberRoleId) {
+			throw Error('Could not find global member role!');
```
Comment (ivov) [inline packages/cli/src/databases/migrations/common/1700571993961-AddGlobalAdminRole.ts:42]:
```suggestion
			throw new Error('Could not find global member role!');
```

File: packages/cli/src/databases/migrations/common/1700571993961-AddGlobalAdminRole.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		const memberRoleIdResult: Array<{ id: number }> = await runQuery(
+			`SELECT id FROM ${roleTableName} WHERE name = :name AND scope = :scope`,
+			{
+				name: 'member',
+				scope: 'global',
+			},
+		);
+
+		const adminRoleId = adminRoleIdResult[0]?.id;
+		if (!adminRoleId) {
```
Comment (ivov) [inline packages/cli/src/databases/migrations/common/1700571993961-AddGlobalAdminRole.ts:34]: This ID can probably never be `0` but ideally we should try not to rely on coercion. Same below.

**Review by ivov — DISMISSED** [review]
LGTM

File: packages/cli/src/databases/migrations/common/1700571993961-AddGlobalAdminRole.ts (hunk 2 above)
Comment (valya) [reply packages/cli/src/databases/migrations/common/1700571993961-AddGlobalAdminRole.ts:42]: I haven't written Python in a while but I still manage to do this :facepalm:

**Review by ivov — APPROVED** [review]

**Comment by luizeof** [comment]
@Joffcom, so can we have 2 admins in the community installation?

**Comment by Joffcom** [comment]
@luizeof no idea buddy this is probably part of the new roles that is on the way.

**Comment by krynble** [comment]
@luizeof yes we are planning on introducing the new admin role, which is similar to the instance owner, but we will keep a single owner, effectively granting managing permissions to more users. Stay tuned :)

**Comment by krynble** [comment]
Sorry @luizeof I re-read your comment and it's a new feature we're building but will most probably come to enterprise plans only :( I read only the part about having 2 admins.

**Comment by janober** [comment]
Got released with [`n8n@1.19.4`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.19.4)

**Pull request merged**